src/gentypes.jl

```diff
@@ -285,10 +281,7 @@ end
 Evaluate the result of the [`generatetypes`](@ref) function in the current scope.
 """
 macro generatetypes(json_str, module_name)
-    return quote
-        local mod = JSON3.generatetypes($(esc(json_str)), $(esc(module_name)))
-        return Core.eval($__module__, mod)
-    end
+    :(Core.eval($__module__, generatetypes($(esc(json_str)), $(esc(module_name)))))
```

Comment:
Was the `JSON3.` prefix causing an issue? Or it's just not needed?

Reply:
That's my primary theory.  Though it worked fine in use, so it must be something with the interaction of @testset and it.

Here's what the `@macroexpand` looks like without and then with the `JSON3.`
```julia
julia> @macroexpand JSON3.@generatetypes("""{"a": 1}""")
:((JSON3.Core).eval(Main, JSON3.generatetypes("{\"a\": 1}", :JSONTypes)))

julia> @macroexpand JSON3.@generatetypes("""{"a": 1}""")
:((JSON3.Core).eval(Main, (JSON3.JSON3).generatetypes("{\"a\": 1}", :JSONTypes)))
```